### PR TITLE
SDK, CLI examples for phi-3-mini

### PR DESCRIPTION
**Pull request by shubhamiit**
# Description

SDK, CLI examples for phi-3-mini-v.

# Checklist

- [x] I have read the contribution guidelines.
  - [General](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
  - [SDK](https://github.com/Azure/azureml-examples/blob/main/sdk/CONTRIBUTING.md)
  - [CLI](https://github.com/Azure/azureml-examples/blob/main/cli/CONTRIBUTING.md)
- [x] I have coordinated with the docs team (mldocs@microsoft.com) if this PR deletes files or changes any file names or file extensions.
- [x] Pull request includes test coverage for the included changes.
- [x] This notebook or file is added to the [CODEOWNERS](https://github.com/Azure/azureml-examples/blob/main/.github/CODEOWNERS) file, pointing to the author or the author's team.

### CODE REVIEW

**Comment by shubhamiit** [comment]
SKU to NC48 and model name should be updated.

File: cli/foundation-models/system/inference/phi-3-mini-v/phi-3-mini-v-128k-instruct/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+$schema: https://azuremlschemas.azureedge.net/latest/managedOnlineDeployment.schema.json
+name: demo
+instance_type: Standard_NC24ads_A100_v4
```
Comment (rjaincc) [inline cli/foundation-models/system/inference/phi-3-mini-v/phi-3-mini-v-128k-instruct/deploy.yml:3]: Update the SKU

File: cli/foundation-models/system/inference/phi-3-mini-v/phi-3-mini-v-128k-instruct/online-endpoint-deployment.sh [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+# script inputs
+subscription_id="<SUBSCRIPTION_ID>"
+resource_group_name="<RESOURCE_GROUP>"
+workspace_name="<WORKSPACE_NAME>"
+
+# This is the model from system registry that needs to be deployed
+registry_name="azureml"
+model_name="Phi-3-mini-v-128k-instruct-patched-vllm"
+deployment_sku="Standard_NC24ads_A100_v4"
```
Comment (rjaincc) [inline cli/foundation-models/system/inference/phi-3-mini-v/phi-3-mini-v-128k-instruct/online-endpoint-deployment.sh:11]: Update the SKU

File: cli/foundation-models/system/inference/phi-3-mini-v/phi-3-mini-v-128k-instruct/online-endpoint-deployment.sh [hunk 3]
```diff
@@ -0,0 +1,79 @@
+set -x
+
+# script inputs
+subscription_id="<SUBSCRIPTION_ID>"
+resource_group_name="<RESOURCE_GROUP>"
+workspace_name="<WORKSPACE_NAME>"
+
+# This is the model from system registry that needs to be deployed
+registry_name="azureml"
+model_name="Phi-3-mini-v-128k-instruct-patched-vllm"
```
Comment (rjaincc) [inline cli/foundation-models/system/inference/phi-3-mini-v/phi-3-mini-v-128k-instruct/online-endpoint-deployment.sh:10]: Use the right model name

File: sdk/python/foundation-models/system/inference/phi-3-mini-v/phi-3-mini-v-128k-instruct/online-endpoint-chat-completions-inference.ipynb [hunk 4]
```diff
@@ -0,0 +1,257 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Phi 3 Mini 128k Instruct inference using Online Endpoints\n",
+    "\n",
+    "This sample shows how to deploy `phi-3-mini-v-128k-instruct` to an online endpoint for inference.\n",
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/inference/phi-3-mini-v/phi-3-mini-v-128k-instruct/online-endpoint-chat-completions-inference.ipynb:10]: Use the right Final model name

File: sdk/python/foundation-models/system/inference/phi-3-mini-v/phi-3-mini-v-128k-instruct/online-endpoint-chat-completions-inference.ipynb [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+    "Online endpoints give a durable REST API that can be used to integrate with applications that need to use the model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model_name = \"Phi-3-mini-v-128k-instruct-patched-vllm\"\n",
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/inference/phi-3-mini-v/phi-3-mini-v-128k-instruct/online-endpoint-chat-completions-inference.ipynb:76]: Use the right final model name

File: sdk/python/foundation-models/system/inference/phi-3-mini-v/phi-3-mini-v-128k-instruct/online-endpoint-chat-completions-inference.ipynb [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# create a deployment\n",
+    "demo_deployment = ManagedOnlineDeployment(\n",
+    "    name=\"demo\",\n",
+    "    endpoint_name=online_endpoint_name,\n",
+    "    model=foundation_model.id,\n",
+    "    instance_type=\"Standard_NC24ads_A100_v4\",\n",
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/inference/phi-3-mini-v/phi-3-mini-v-128k-instruct/online-endpoint-chat-completions-inference.ipynb:128]: Update the SKU

File: cli/foundation-models/system/inference/visual-chat-completion/online-endpoint-deployment.sh [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+# script inputs
+subscription_id="<SUBSCRIPTION_ID>"
+resource_group_name="<RESOURCE_GROUP>"
+workspace_name="<WORKSPACE_NAME>"
+
+# This is the model from system registry that needs to be deployed
+registry_name="azureml"
+model_name="Phi-3-vision-128k-instruct"
+deployment_sku="phi-3-vision-128k-instruct"
```
Comment (svaruag) [inline cli/foundation-models/system/inference/visual-chat-completion/online-endpoint-deployment.sh:11]: Standard_NC48ads_A100_v4

**Review by nvijayrania — APPROVED** [review]

File: cli/foundation-models/system/inference/phi-3-mini-v/phi-3-mini-v-128k-instruct/online-endpoint-deployment.sh (hunk 3 above)
Comment (nvijayrania) [reply cli/foundation-models/system/inference/phi-3-mini-v/phi-3-mini-v-128k-instruct/online-endpoint-deployment.sh]: Shouldn't the mode name be "Phi-3-vision-128k-instruct"
https://ml.azure.com/registries/azureml-preview/models/Phi-3-vision-128k-instruct/version/1

File: sdk/python/foundation-models/system/inference/phi-3-mini-v/phi-3-mini-v-128k-instruct/online-endpoint-chat-completions-inference.ipynb (hunk 5 above)
Comment (nvijayrania) [reply sdk/python/foundation-models/system/inference/phi-3-mini-v/phi-3-mini-v-128k-instruct/online-endpoint-chat-completions-inference.ipynb]: Discussed with @rjaincc. model name should be "Phi-3-vision-128k-instruct".

**Review by gauravrajguru — APPROVED** [review]

**Pull request merged**